src/color.rs

```diff
@@ -267,28 +365,38 @@ impl ColorSpace<'_> {
 }
 
 /// Special color spaces.
-impl ColorSpace<'_> {
-    /// Write a `Separation` color space. PDF 1.2+.
-    pub fn separation(self, color_name: Name, base: Name, tint: Ref) {
+impl<'a> ColorSpace<'a> {
+    /// Start writing a `Separation` color space. PDF 1.2+.
+    ///
+    /// The `color_name` argument is the name of the colorant that will be
+    /// used by the printer.
+    pub fn separation(self, color_name: Name) -> Separation<'a> {
         let mut array = self.obj.array();
         array.item(ColorSpaceType::Separation.to_name());
         array.item(color_name);
-        array.item(base);
-        array.item(tint);
+        Separation::start(array)
     }
 
     /// Write a `DeviceN` color space. PDF 1.3+.
+    ///
+    /// The `names` argument contains the N names of the color components and
+    /// the respective colorants. The `alternate_space` argument describes an
+    /// alternate color space to use if the color names are unknown. The `tint`
+    /// argument is an indirect reference to a function that maps from an n-
+    /// dimensional values with components between 0 and 1 to a color in the
+    /// alternate color space.
```

Comment:
shouldn't device n work the same way as separation? let alternate_space and tint be written directly?